src/BuildingRegistry/Building/Building_BuildingUnit.cs

```diff
@@ -171,7 +171,28 @@ public void CorrectRealizeBuildingUnit(BuildingUnitPersistentLocalId buildingUni
             EnsureCommonBuildingUnit(addCommonBuildingUnit);
         }
 
-        public void CorrectPositionBuildingUnit(
+        public void CorrectBuildingUnitRemoval(
+            IAddCommonBuildingUnit addCommonBuildingUnit,
+            BuildingUnitPersistentLocalId buildingUnitPersistentLocalId)
+        {
+            GuardRemovedBuilding();
+            GuardBuildingValidStatuses(BuildingStatus.Planned, BuildingStatus.UnderConstruction, BuildingStatus.Realized);
+
+            var buildingUnit = _buildingUnits
+                .GetByPersistentLocalId(buildingUnitPersistentLocalId);
+
+            if ((BuildingStatus == BuildingStatus.Planned || BuildingStatus == BuildingStatus.UnderConstruction)
+                && buildingUnit.Status == BuildingUnitStatus.Realized)
+            {
+                throw new BuildingUnitHasInvalidStatusException();
```

Comment:
Shouldn't we then move it to planned? Otherwise it's impossible to do.

Reply:
Yeah, that's the question I posed to Koen. It will be necessary to first realize the building before correcting the removal. But that could possibly be an undesired action. I'll change it and wait for Koen for the confirmation.